src/couch_replicator_connection.erl

```diff
@@ -171,6 +175,33 @@ delete_worker(Worker) ->
     ok.
 
 
+-spec incr_monitor(pid(), #state{}) -> #state{}.
+incr_monitor(Owner, State) ->
+    NewMonitors = case dict:find(Owner, State#state.monitors) of
+        error ->
+            MRef = monitor(process, Owner),
+            dict:store(Owner, {1, MRef}, State#state.monitors);
```

Comment:
I don't understand how that'd work. Like, if you have an owner acquire workers A, B, and C (in that order) but then relinquished them in order A, B, C, it'd demonitor after it relinquished A, right? Leaving the owner unmonitored while B and C were still owned, right?